haystack/document_stores/opensearch.py

```diff
@@ -613,14 +684,15 @@ def _get_vector_similarity_query(self, query_emb: np.ndarray, top_k: int):
     def _get_raw_similarity_score(self, score):
         # adjust scores according to https://opensearch.org/docs/latest/search-plugins/knn/approximate-knn
         # and https://opensearch.org/docs/latest/search-plugins/knn/knn-score-script/
-        if self.similarity == "dot_product":
+        space_type = self.similarity_to_space_type[self.knn_engine][self.similarity]
+        if space_type == "innerproduct":
```

Comment:
According to the docs there should be a different calculation for scriptscoring. Check this before merging:
https://opensearch.org/docs/latest/search-plugins/knn/approximate-knn/
https://opensearch.org/docs/latest/search-plugins/knn/knn-score-script/

Reply:
The impl here is correct, instead it's an issue with the OpenSearch docs, see https://github.com/opensearch-project/documentation-website/issues/1258